File: LeetCode/078.subsets/subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums){
        List<List<Integer>> res = new ArrayList<>(1 << nums.length);
        for (int i = 0;i < (1 << nums.length);i++){
            List<Integer>tmp = new ArrayList<>(nums.length);
            for (int j = 0;j < nums.length;j++){
                if ((i & (1 << j)) != 0){
                    tmp.add(nums[j]);
                }
            }
            res.add(tmp);
        }
        return res;
    }
}

File: LeetCode/0078.subsets/subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>>res;
        int size = nums.size();
        for (int i = 0; i < (1 << size); i++){
            vector<int>itv;
            for (int j = 0; j < size; j++){
                if ((i & (1 << j)) != 0){
                    itv.push_back(nums[j]);
                }
            }
            res.push_back(itv);
        }
        return res;
    }
};